akka-remote/src/main/scala/akka/remote/artery/tcp/SSLEngineProvider.scala

```diff
@@ -84,19 +85,37 @@ class ConfigSSLEngineProvider(protected val config: Config, protected val log: M
   val SSLRequireMutualAuthentication: Boolean = config.getBoolean("require-mutual-authentication")
   val HostnameVerification: Boolean = config.getBoolean("hostname-verification")
 
-  private lazy val sslContext: SSLContext = {
-    // log hostname verification warning once
-    if (HostnameVerification)
-      log.debug("TLS/SSL hostname verification is enabled.")
-    else
-      log.info(
-        LogMarker.Security,
-        "TLS/SSL hostname verification is disabled. See Akka reference documentation for more information.")
-
-    constructContext()
+  private val SSLContextCacheTime: FiniteDuration = config.getDuration("ssl-context-cache-ttl").toMillis.millis
+
+  // log hostname verification warning once
+  if (HostnameVerification)
+    log.debug("TLS/SSL hostname verification is enabled.")
+  else
+    log.info(
+      LogMarker.Security,
+      "TLS/SSL hostname verification is disabled. See Akka reference documentation for more information.")
+
+  private def sslContext: SSLContext = getFromCache
+
+  private val contextRef = new AtomicReference[Option[CachedSSLContext]](None)
+
+  private case class CachedSSLContext(cached: SSLContext, expires: Deadline)
+
+  private def getFromCache: SSLContext = {
+    contextRef.get() match {
+      case Some(CachedSSLContext(oldContext, expired)) if expired.isOverdue() =>
+        val context = constructContext()
+        contextRef.set(Some(CachedSSLContext(context, SSLContextCacheTime.fromNow)))
+        context
+      case Some(CachedSSLContext(cached, _)) => cached
+      case None =>
+        val context = constructContext()
```

Comment:
would it be better to make `getFromCache` `synchronized` instead of `constructContext`, because now several threads may block here and then create the context again when released